Backend

PersianBlue opened the pull request:
Project essentially complete, pending review and suggestions

PersianBlue committed:
Added background colours and borders

PersianBlue committed:
Finalizing style, almost done

PersianBlue committed:
More CSS tweaks, just need bevels and grooves

PersianBlue committed:
Added drop shadows for 3d effects

PersianBlue committed:
Merge branch 'main' into Backend

PersianBlue committed:
Fine tuned the noisy gradient background

PersianBlue committed:
Adding explanations in comments

PersianBlue committed:
Exported inline css to files and tidied up positions

PersianBlue committed:
Hardcoded plus and minus signs from screenshots

PersianBlue committed:
Added glow on hover effect to all buttons

Natreve commented on src/components/button/index.js:
```diff
@@ -1,7 +1,16 @@
+  //const styles = []
+    return (
+    <div>
+      <button className = {css.circular} id = {props.id} onClick = {props.onClick}>{props.children}</button>
```
You can do the following on the class declaration
`<Button type="roundedEdges" />`
For the component itself you can now do
`<button className={css[props.type]} />`

PersianBlue replied:
Woah! I was able to transform like 6 if statements into two conditionals lol maaan. I was looking for something like this before because I didn't want to write so many conditionals, but couldn't find anything. Thanks a million for this

Natreve commented on src/components/button/css.module.scss:
```diff
@@ -13,3 +13,38 @@
+
+
+
+#changeSeconds{
```
Why are you using this?

PersianBlue replied:
I've adjusted the come somewhat from before this pull request, so the lines don't match up as much with my code anymore, but if you mean the roundedEdges class, it's to make the + and - buttons which are square with rounded edges

Natreve commented on src/components/team/css.module.scss:
```diff
@@ -0,0 +1,38 @@
+    background: rgb(179, 229, 248);
+}
+
+.roundedEdges{
```
Noticed you use this more than once for different classes this should be in the general scss file. Try to always avoid duplicating code.

PersianBlue replied:
Gotchu

Natreve commented on src/components/throwtimer/css.module.scss:
```diff
@@ -0,0 +1,27 @@
+    //color: red;
+  }
+
+  .circular{
```
Any css class you find yourself reusing more than once suggest that they need to be placed in a global file to be access by all components instead of being recreated more than once.

PersianBlue replied:
Ahh that makes sense ty

Natreve commented on src/pages/index.js:
```diff
@@ -9,11 +9,13 @@ import "../global.scss"
-      <Team name = "Team One" id ={1}/>
-      <Team name = "Team Two" id = {2}/>
-      <ThrowTimer/>
+      <container style={{width: 350}}>
```
There is no such thing as a container tag to my knowledge it's best to not create your own html tags, but instead use what is available like divs, or create a component called Container

PersianBlue replied:
Lol somehow I assumed that containers that I heard people talking about were made with a container tag lol. Back to divs it is

PersianBlue committed:
Condensed button code & css

Condensed if statements in button component to simpler version
Removed some CSS that was no longer needed to make the code work

PersianBlue committed:
removing white space

Natreve reviewed (commented):
png file are unnecessarily large, you can actually use css to create icons. Alternatively you can use svg. But for now this is fine.
